src/openzeppelin/token/erc1155/presets/ERC1155MintableBurnable.cairo

```diff
@@ -0,0 +1,252 @@
+# SPDX-License-Identifier: MIT
+# OpenZeppelin Contracts for Cairo v0.2.1 (token/erc1155/presets/ERC1155MintableBurnable.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.uint256 import Uint256
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.math import assert_not_zero
+
+from openzeppelin.access.ownable.library import Ownable
+from openzeppelin.token.erc1155.library import ERC1155
+from openzeppelin.introspection.erc165.library import ERC165
+
+#
+# Constructor
+#
+
+@constructor
+func constructor{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(uri: felt, owner: felt):
+    ERC1155.initializer(uri)
+    Ownable.initializer(owner)
+    return ()
+end
+
+#
+# Getters
+#
+
+@view
+func supportsInterface{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(interfaceId: felt) -> (success: felt):
+    return ERC165.supports_interface(interfaceId)
+end
+
+@view
+func uri{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }() -> (uri: felt):
+    return ERC1155.uri()
+end
+
+@view
+func balanceOf{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(account: felt, id: Uint256) -> (balance: Uint256):
+    return ERC1155.balance_of(account, id)
```

Comment:
does this really work? then why are we doing this everywhere?
```cairo
let (is_approved) = ERC1155.is_approved_for_all(account, operator)
return (is_approved)
```

Reply:
This does work, and should in general as long as the implicit args and return type (including the names) match. The name of the returned value of `isApprovedForAll` is different from that of its implementation in the library (`isApproved` vs `is_approved`), hence the use of the let expression. I must have missed `balanceOf` when I changed things for consistency.